src/components/OpenWith/OpenWith.js

```diff
@@ -192,9 +200,64 @@ class OpenWith extends PureComponent<Props, State> {
      * @private
      * @return {void}
      */
-    onIntegrationClick(): void {
-        /* no-op */
-    }
+    onIntegrationClick = (appIntegrationId: string): void => {
+        const { fileId }: Props = this.props;
+        this.api
+            .getAppIntegrationsAPI(false)
+            .execute(
+                appIntegrationId,
+                fileId,
+                this.executeIntegrationSuccessHandler,
+                this.executeIntegrationErrorHandler,
+            );
+    };
+
+    /**
+     * Opens the integration in a new tab based on the API data
+     *
+     * @private
+     * @param {ExecuteAPI} executeData - API response on how to open an executed integration
+     * @return {void}
+     */
+    executeIntegrationSuccessHandler = (executeData: ExecuteAPI) => {
+        switch (executeData.method.toLowerCase()) {
+            case HTTP_POST:
+                this.setState({ executePostData: executeData }, () => {
+                    const form: ?HTMLFormElement = document.querySelector(
+                        `#bcow-execute-form-${this.id}`,
+                    );
+
+                    if (!form) {
+                        this.executeIntegrationErrorHandler(
+                            Error('execution form unavailable'),
+                        );
+                        return;
+                    }
+
+                    form.submit();
+                });
+
+                break;
+            case HTTP_GET:
+                window.open(executeData.url, '_blank');
```

Comment:
Consider handling the case where the window wasn't openable, which should return `null` here.